coralproject/talk
# Add a `secretBadges` array to the user model

## What does this PR do?

Adds a `secretBadges` array to the user model which is only configurable via Mongo and used for special cases by the organization.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

N/A

## How do I test this PR?

- use multi-site test with SSO users
- find an SSO user in Mongo, add a `secretBadges` array field to their user document and add an entry (they are string values in an array)
- visit the site and post a comment as that user via SSO
- see that the user has the "secret badge" you gave them in Mongo
- see that any other SSO badges also exist

## Were any tests migrated to React Testing Library?

No

## How do we deploy this PR?

If we want to formally release this, rebase this off of `develop` (instead of v8.7.1) and merge it as usual into `develop`.
